src/entities/swap/types.ts

```diff
@@ -1,14 +1,26 @@
 import { TokenAmount } from '../tokenAmount';
-import { MinimalToken, Slippage } from '../..';
-import { SingleSwap, SwapKind, BatchSwapStep } from '../../types';
-import { PathWithAmount } from './pathWithAmount';
+import { Slippage } from '../..';
+import { SwapKind } from '../../types';
 import { Address, Hex } from 'viem';
-import {
-    SwapCallBuildV2,
-    SwapCallExactInV2,
-    SwapCallExactOutV2,
-} from './swapV2';
-import { SingleTokenExactIn, SingleTokenExactOut } from './swapV3';
+import { SwapCallBuildInputBaseV2 } from './swaps/v2';
+import { Path } from './paths/types';
+
+export type SwapInput = {
+    chainId: number;
+    paths: Path[];
+    swapKind: SwapKind;
+};
+
+type SwapCallBuildInputBase = {
+    deadline?: bigint;
+    slippage: Slippage;
+    wethIsEth?: boolean;
+    queryOutput: ExactInQueryOutput | ExactOutQueryOutput;
+};
+
+export type SwapCallBuildInput =
+    | SwapCallBuildInputBase
+    | (SwapCallBuildInputBase & SwapCallBuildInputBaseV2);
```

Comment:
Yep, that was actually annoying me tbh! Nice suggestion about limits. I've refactored quite a bit, there's some code repitition but I think its a good bit more intuitive and easier to follow but let me know what you think?

I'm not sure what the benefits would be from making the main `Swap` object implement the `SwapBase`. It would be easier to do after this refactor but would mostly be adding redundant properties/functions that I think would clutter unnecessarily. Let me know if you think thats the wrong conclusion though.

Reply:
One possible solution would be to break SwapBase into SwapBase (methods) and SwapState (attributes), and have the external Swap implement only SwapBase.
But I'd say the current code is is no longer confusing, so we should be good to go with it 👍